pkgs/top-level/ocaml-packages.nix

```
@@ -357,7 +357,12 @@ let
       then pkgs.dune_2
       else throw "dune_2 is not available for OCaml ${ocaml.version}";
 
-    dune_3 = callPackage ../development/tools/ocaml/dune/3.nix { };
+    dune_3 =
+      if lib.versionAtLeast ocaml.version "4.08"
+      then callPackage ../development/tools/ocaml/dune/3.nix { }
+      else if lib.versionAtLeast ocaml.version "4.02"
```

```suggestion
      else if lib.versionAtLeast ocaml.version "4.02.3"
```

As per the README mention. Can you also update the dune_2 upper.

---

this is probably not needed given that nixpkgs already is on 4.02.3 for the 4.02 compiler.